pkg/build/api/v1/types.go

```diff
@@ -303,6 +303,10 @@ type BuildStrategy struct {
 
 	// CustomStrategy holds the parameters to the Custom build strategy
 	CustomStrategy *CustomBuildStrategy `json:"customStrategy,omitempty"`
+
+	// JenkinsPipelineStrategy holds the parameters to the Jenkins Pipeline build strategy.
+	// This strategy is experimental.
```

Comment:
@smarterclayton do we need to do anything else here to warn users not to use this?  the expectation is we're going to merge this branch (thus exposing the strategy) before we have the rest of the pieces (mainly the jenkins image) in place to be able to actually run/manage these builds correctly.